### PR TITLE
Remove .length from variable $pin_release in app

### PR DESCRIPTION
**Pull request by paladox**
This causes 

Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Unknown function: 'length'. at /etc/puppet/environments/production/modules/apt/templates/pin.pref.epp:4:16 at /etc/puppet/environments/production/modules/apt/manifests/source.pp:92 on node xxx

downstream

We use puppet from Debian stretch

### CODE REVIEW

**Comment by tphoney** [comment]
closing, and re-openning to force travis to re-kick

File: templates/pin.pref.epp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <%- | $name, $pin_release, $release, $codename, $release_version, $component, $originator, $label, $version, $origin, $explanation, $packages_string, $priority | -%>
 <%-
 $pin =
-if $pin_release.length > 0 {
+if $pin_release {
```
Comment (tphoney) [inline templates/pin.pref.epp]: would this be better
`if $pin_release and $pin_release.length > 0`
Following the logic of the rest of the template.
Many thanks for the code !!! @paladox

Comment (paladox) [reply templates/pin.pref.epp]: Thanks, done.

**Comment by tphoney** [comment]
thanks @paladox i will have another look, and get back to you.

**Comment by tphoney** [comment]
apologies, for the interruption again, what manifest/data where you using to test this ?

**Comment by tphoney** [comment]
or revert back to what you had, and i will merge.

**Comment by paladox** [comment]
@tphoney we have this:

    apt::pin { 'debian_stretch_backports_redis':
        priority   => 740,
        originator => 'Debian',
        release    => 'stretch-backports',
        packages   => 'redis-server',
    }

https://github.com/miraheze/puppet/blob/master/modules/redis/manifests/init.pp#L9

https://github.com/miraheze/puppet/blob/master/manifests/site.pp#L42

**Comment by paladox** [comment]
doing

```
if $pin_release {
  $options = [
    if $release != ''         { "a=${release}" },
    if $codename != ''        { "n=${codename}" },
    if $release_version != '' { "v=${release_version}"},
    if $component != ''       { "c=${component}" },
    if $originator != ''      { "o=${originator}" },
    if $label != ''           { "l=${label}" },
    ].filter |$x| { $x != undef }
   "release ${options.join(', ')}" }
```

should also work?

I think .length will only work on a array?

**Comment by tphoney** [comment]
https://forge.puppet.com/puppetlabs/stdlib#length Its a weird one :)  depends on the puppet version 4 vs 5. I think, revert to 'if $pin_release ' and i will merge.

File: templates/pin.pref.epp [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
```
Comment (paladox) [inline templates/pin.pref.epp]: Should this be != ''?

**Comment by tphoney** [comment]
This looks good™️ 👍. Thanks for the effort you have put in.

**Comment by paladox** [comment]
your welcome and thanks for merging! :)

**Pull request merged**